resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/BulkheadAspect.java

```diff
@@ -61,6 +68,16 @@ public Object bulkheadAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, Bulk
 		}
 		String backend = backendMonitored.name();
 		io.github.resilience4j.bulkhead.Bulkhead bulkhead = getOrCreateBulkhead(methodName, backend);
+		Class<?> returnType = method.getReturnType();
+		if (bulkHeadAspectExts != null && !bulkHeadAspectExts.isEmpty()) {
+			for (BulkHeadAspectExt bulkHeadAspectExt : bulkHeadAspectExts) {
+				if (bulkHeadAspectExt.canHandleReturnType(returnType)) {
+					return bulkHeadAspectExt.handle(proceedingJoinPoint, bulkhead, methodName);
+				}
+			}
+		} else if (CompletionStage.class.isAssignableFrom(returnType)) {
+			return handleJoinPointCompletableFuture(proceedingJoinPoint, bulkhead, methodName);
+		}
 		return handleJoinPoint(proceedingJoinPoint, bulkhead, methodName);
```

Comment:
Should we rename it to handleSyncJoinPoint?

Reply:
I would keep it like that for now .